### PR TITLE
Add a section about token-based authentication

### PR DESCRIPTION
**Pull request by mjrlgue**
It took me a considerable time to figure out the code for a token-based authentication, and I said why not add it to the documentation.

### CODE REVIEW

File: x-pack/docs/en/watcher/input/http.asciidoc [hunk 1]
```diff
@@ -139,6 +139,28 @@ http://openweathermap.org/appid[OpenWeatherMap] service:
 --------------------------------------------------
 // NOTCONSOLE
 
+===== Using Token-Based Authentication
```
Comment (abdonpijpelink) [inline x-pack/docs/en/watcher/input/http.asciidoc:142]:
```suggestion
===== Using token-based authentication
```

**Review by abdonpijpelink — COMMENTED** [review]
This is great @mjrlgue . Thanks for sharing!

I've left some minor suggestions. Also, this PR should target the `main` branch so this change will also be in future versions of the documentation. Do you need a hand rebasing this PR?

File: x-pack/docs/en/watcher/input/http.asciidoc (hunk 1 above)
Comment (mjrlgue) [reply x-pack/docs/en/watcher/input/http.asciidoc:142]: Thanks @abdonpijpelink,  the minor changes you suggested are fine for me. For the rebase sure as it's my first time doing it  ^^

**Comment by abdonpijpelink** [comment]
@elasticmachine test this please

**Review by abdonpijpelink — APPROVED** [review]

**Pull request merged**